src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,299 @@
+                                            incomingEdgeName.charAt(incomingEdgeNameIndex - 1),
+                                            startingEdgeName.charAt(startingEdgeNameIndex - 1)),
+                            results[incomingEdgeNameIndex - 1][startingEdgeNameIndex] + 1,
+                            results[incomingEdgeNameIndex][startingEdgeNameIndex - 1] + 1);
+                }
+            }
+        }
+        // If there's only a single character difference and that character is a directional OR
+        // numerical character, we consider both roads to be different and so we don't flag them.
+        // Else we return the Levenshtein distance as usual.
+        return (possibleDirectionalDifference || singleNumericalDifference)
+                && results[incomingEdgeName.length()][startingEdgeName.length()] == 1 ? -1
+                        : results[incomingEdgeName.length()][startingEdgeName.length()];
+    }
+
+    /**
+     * Check if the parameter characters are different directional characters.
+     *
+     * @param incomingEdgeCharacter
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if the parameter characters are both directional (members of Direction enum) AND
+     *         they are different from one another, false otherwise
+     */
+    private static boolean hasDirectionalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return DIRECTIONS.stream().map(Enum::toString).filter(
+                equalCharacter(incomingEdgeCharacter).or(equalCharacter(startingEdgeCharacter)))
+                .count() >= 2;
+    }
+
+    /**
+     * Check if the parameter characters are different numerical characters.
+     *
+     * @param incomingEdgeCharacter
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if both parameter characters are different numbers in [0,9], false otherwise
+     */
+    private static boolean hasNumericalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return Character.isDigit(incomingEdgeCharacter) && Character.isDigit(startingEdgeCharacter)
+                && IntStream.range(ZERO, NINE)
+                        .filter(number -> number == (incomingEdgeCharacter - ASCII_OFFSET)
+                                || number == (startingEdgeCharacter - ASCII_OFFSET))
+                        .count() >= 2;
+    }
+
+    /**
+     * Retrieve the minimum value out of parameter numbers.
+     *
+     * @param numbers
+     *            the numbers on which to operate
+     * @return the minimum of those numbers
+     */
+    private static int min(final int... numbers)
+    {
+        return Arrays.stream(numbers).min().orElse(Integer.MAX_VALUE);
```

Comment:
I know we're pulling in at least part of the guava library already. If we're already pulling in guava's `Ints` class (I don't know for sure if we are or not), we could replace all of the calls to this method with [Ints.min()](https://guava.dev/releases/snapshot/api/docs/com/google/common/primitives/Ints.html#min-int...-)